### PR TITLE
[feat] Add ability to define properties on Container subclasses

### PR DESCRIPTION
**Pull request by alberttorosyan**
Support for ORM-like properties definition:

```python
class Run(Container):
  name = Property()
  archived = Property(default=False)
```

would result to creation of name and archived attributes for Run instances, as well as handle serialization/deserialization in rocksdb.

On the UI SDK side, while parsing the API call result stream, a special keyword `$property` is used to determine the list of properties. These properties can be accessed as attributes, similar to the Python SDK.

### CODE REVIEW

File: src/python/aim/__init__.py [hunk 1]
```diff
@@ -2,7 +2,7 @@
 
 from .record import Record
 from .sequence import Sequence
-from .container import Container
+from .container import Container, Property
```
Comment (mihran113) [inline src/python/aim/__init__.py:5]: why won't include the `Property` in `__all__` as well?

**Review by mihran113 — APPROVED** [review]
left some minor comments, other than that looks great

File: pkgs/aimstack/asp/models/run.py [hunk 2]
```diff
@@ -41,6 +41,10 @@
 
 @type_utils.query_alias('run')
 class Run(Container, Caller):
+    name = Property()
```
[the remark this answers is not in the document]
Comment (alberttorosyan) [reply pkgs/aimstack/asp/models/run.py:44]: I'd suggest to open a GH issue

File: src/python/aim/__init__.py (hunk 1 above)
Comment (alberttorosyan) [reply src/python/aim/__init__.py:5]: good catch. will make changes soon

File: pkgs/aimstack/asp/models/run.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (alberttorosyan) [reply pkgs/aimstack/asp/models/run.py:44]: GH issue opened: https://github.com/aimhubio/aim/issues/2974

**Review by roubkar — APPROVED** [review]

**Pull request merged**